client/src/pages/Login.jsx

```diff
@@ -1,141 +1,79 @@
 import { useMutation } from "@apollo/client";
-import React, { useState } from "react";
+import React from "react";
 import { Link } from "react-router-dom";
 import { LOGIN } from "../utils/mutations";
 import Auth from "../utils/auth";
-import {
-  Button,
-  Sheet,
-  Typography,
-  FormControl,
-  FormLabel,
-  Input,
-} from "@mui/joy";
+import { Button, Typography, Input, Card, FormHelperText } from "@mui/joy";
+import { useForm } from "react-hook-form";
+import { useSchemas } from "../hooks/useSchemas";
+import { yupResolver } from "@hookform/resolvers/yup";
 
-const Login = () => {
-  const [formData, setFormData] = useState({
-    email: "",
-    password: "",
-  });
-
-  const [formErrors, setFormErrors] = useState({
-    email: "",
-    password: "",
-  });
-
-  const handleChange = (event) => {
-    const { name, value } = event.target;
-    setFormData((prevData) => ({ ...prevData, [name]: value }));
-  };
+const styles = {
+  card: {
+    width: 400,
+    mx: "auto",
+    p: 2,
+    mt: 4,
+    my: 4,
+    borderRadius: "sm",
+    display: "flex",
+    flexDirection: "column",
+    gap: 2,
+    boxShadow: "lg",
+    backgroundColor: "lightblue",
+  },
+};
 
-  const [login, { errors }] = useMutation(LOGIN);
-  const validateForm = () => {
-    let errors = {};
-    let isValid = true;
+const Login = () => {
+  const [login, { error }] = useMutation(LOGIN);
 
-    if (!formData.email.trim()) {
-      errors.email = "Please enter your email";
-      isValid = false;
-    }
+  const { loginSchema } = useSchemas();
 
-    if (!formData.password) {
-      errors.password = "Please enter your password";
-      isValid = false;
-    }
-    setFormErrors(errors);
-    return isValid;
-  };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(loginSchema) });
 
-  const handleSubmit = async (event) => {
-    console.log("in handlesubmit");
-    event.preventDefault();
-    if (validateForm()) {
-      try {
-        const { data } = await login({
-          variables: { ...formData },
-        });
-        Auth.login(data);
-        alert("Successfully logged in!");
-      } catch (err) {
-        console.error(err);
-        alert(err);
-        alert("Login failed. Please try again.");
-      }
-      setFormData({
-        email: "",
-        password: "",
+  const onSubmit = async (userInput) => {
+    try {
+      const { data } = await login({
+        variables: { ...userInput },
       });
+      Auth.login(data);
+      alert("Successfully logged in!");
```

Comment:
So does `login` here always return the happy path? What if the credentials are wrong? I gather that there's where the `error` object from the request comes in, but what do we pass to `Auth.login` in that case?

Reply:
If the wrong credentials are passed in it will hit the resolver in the server and throw a new error. However, it doesn't look like I am passing that error back to the front-end to inform the user. It is showing up in the console. I'll have to work on that.